tests/vmcheck/test-autoupdate.sh

```diff
@@ -0,0 +1,197 @@
+lift_commit() {
+  checksum=$1; shift
+  branch=$1; shift
+  vm_cmd ostree pull-local --repo=$REMOTE_OSTREE --disable-fsync \
+    /ostree/repo $checksum
+  vm_cmd ostree --repo=$REMOTE_OSTREE refs $branch --delete
+  vm_cmd ostree --repo=$REMOTE_OSTREE refs $checksum --create=$branch
+}
+
+# use a previously stolen commit to create an update on our vmcheck branch,
+# complete with version string and pkglist metadata
+create_update() {
+  branch=$1; shift
+  vm_cmd ostree commit --repo=$REMOTE_OSTREE -b vmcheck \
+    --tree=ref=$branch --add-metadata-string=version=$branch --fsync=no
+  # avoid libtool wrapper here since we're running on the VM and it would try to
+  # cd to topsrcdir/use gcc; libs are installed anyway
+  vm_cmd /var/roothome/sync/.libs/inject-pkglist $REMOTE_OSTREE vmcheck
+}
+
+# (delete ref but don't prune for easier debugging)
+vm_cmd ostree refs --repo=$REMOTE_OSTREE vmcheck --delete
+
+# now let's build some pkgs that we'll jury-rig into a base update
+# this whole block can be commented out for a speed-up when iterating locally
+vm_build_rpm base-pkg-foo version 1.4 release 7
+vm_build_rpm base-pkg-bar
+vm_build_rpm base-pkg-baz version 1.1 release 1
+vm_rpmostree install base-pkg-{foo,bar,baz}
+lift_commit $(vm_get_pending_csum) v1
+vm_rpmostree cleanup -p
+rm -rf $test_tmpdir/yumrepo
+vm_build_rpm base-pkg-foo version 1.4 release 8 # upgraded
+vm_build_rpm base-pkg-bar version 0.9 release 3 # downgraded
+vm_build_rpm base-pkg-boo version 3.7 release 2.11 # added
+vm_rpmostree install base-pkg-{foo,bar,boo}
+lift_commit $(vm_get_pending_csum) v2
+vm_rpmostree cleanup -p
+
+# ok, we're done with prep, now let's rebase on the first revision and install a
+# layered package
+create_update v1
+vm_cmd ostree remote add vmcheckmote --no-gpg-verify http://localhost:8888/
+vm_build_rpm layered-cake version 2.1 release 3
+vm_rpmostree rebase vmcheckmote:vmcheck --install layered-cake
+vm_reboot
+vm_rpmostree status -v
+vm_assert_status_jq \
+    ".deployments[0][\"origin\"] == \"vmcheckmote:vmcheck\"" \
+    ".deployments[0][\"version\"] == \"v1\"" \
+    '.deployments[0]["packages"]|length == 1' \
+    '.deployments[0]["packages"]|index("layered-cake") >= 0'
+echo "ok prep"
+
+# start it up again since we rebooted
+vm_start_httpd ostree_server $REMOTE_OSTREE 8888
+
+change_policy() {
+  policy=$1; shift
+  vm_cmd cp /usr/etc/rpm-ostreed.conf /etc
+  cat > tmp.sh << EOF
+echo -e "[Daemon]\nAutomaticUpdatePolicy=$policy" > /etc/rpm-ostreed.conf
+EOF
+  vm_cmdfile tmp.sh
+  vm_rpmostree reload
+}
+
+# make sure that off means off
+change_policy off
+vm_rpmostree status | grep 'auto updates disabled'
+vm_rpmostree upgrade --automatic > out.txt
+assert_file_has_content out.txt "Automatic updates are not enabled; exiting"
+echo "ok disabled"
+
+# ok, let's test out check
+change_policy check
+vm_rpmostree status | grep 'auto updates enabled (check'
+
+# build an *older version* and check that we don't report an update
```

Comment:
Yes.  But this actually gets into https://github.com/projectatomic/rpm-ostree/issues/1219